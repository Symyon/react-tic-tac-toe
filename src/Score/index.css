.Score {
    width: 100%;
    padding: 1rem;
    border-radius: 1em;
    margin: 1em;
}