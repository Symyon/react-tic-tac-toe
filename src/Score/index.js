import React, { Component } from 'react';
import './index.css';

class Score extends Component {
    render() {
        const {text, active} = this.props;

        return (
            <div className="Score" style={{backgroundColor: active ? 'rgb(235, 235, 2)' : 'rgb(131, 131, 1)'}}>
                {text}
            </div>
        );
    }
}

export default Score;