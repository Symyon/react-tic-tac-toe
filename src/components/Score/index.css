.Score {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    padding: 1rem;
    border-radius: 1em;
    margin: 1em;
}

.Icon {
    width: 2em;
    height: auto;
    padding-left: 2px;
    padding-right: 2px;
}