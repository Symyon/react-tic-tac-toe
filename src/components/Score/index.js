import React, { Component } from 'react';
import './index.css';

class Score extends Component {
    render() {
        const { icon, text, active } = this.props;

        return (
            <div className="Score" style={{ backgroundColor: active ? 'rgb(235, 235, 2)' : 'rgb(131, 131, 1)' }}>
                <img src={icon} alt='icon' className="Icon"/>
                {text}
            </div>
        );
    }
}

export default Score;