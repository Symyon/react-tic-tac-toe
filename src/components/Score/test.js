import React from 'react';
import { shallow } from 'enzyme';
import Score from './';

describe('Score', () => {
    it('render correctly without props', () => {
        const component = shallow(<Score />);

        expect(component).toMatchSnapshot();
    })

    it('render text correctly with given value', () => {
        const component = shallow(<Score value={5} />);
        expect(component).toMatchSnapshot();
    });

    it('render corectly when passing activity flag', () => {
        const activeScore = shallow(<Score active = {true} />);
        expect(activeScore).toMatchSnapshot();
        
        const passiveScore = shallow(<Score active = {false} />);
        expect(passiveScore).toMatchSnapshot();       
    });
});