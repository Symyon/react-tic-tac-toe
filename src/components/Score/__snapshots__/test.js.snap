// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score render corectly when passing activity flag 1`] = `
<div
  className="Score"
  style={
    Object {
      "backgroundColor": "rgb(235, 235, 2)",
    }
  }
>
  <img
    alt="icon"
    className="Icon"
  />
  Score: 0
</div>
`;

exports[`Score render corectly when passing activity flag 2`] = `
<div
  className="Score"
  style={
    Object {
      "backgroundColor": "rgb(131, 131, 1)",
    }
  }
>
  <img
    alt="icon"
    className="Icon"
  />
  Score: 0
</div>
`;

exports[`Score render correctly without props 1`] = `
<div
  className="Score"
  style={
    Object {
      "backgroundColor": "rgb(131, 131, 1)",
    }
  }
>
  <img
    alt="icon"
    className="Icon"
  />
  Score: 0
</div>
`;

exports[`Score render text correctly with given value 1`] = `
<div
  className="Score"
  style={
    Object {
      "backgroundColor": "rgb(131, 131, 1)",
    }
  }
>
  <img
    alt="icon"
    className="Icon"
  />
  Score: 5
</div>
`;
